### PR TITLE
Rename NonLocalObjectReference to PodReference, and add UID

### PR DESCRIPTION
**Pull request by timstclair**
@pwittrock @vishh

### CODE REVIEW

File: pkg/kubelet/dockertools/labels.go [hunk 1]
```diff
@@ -73,6 +73,10 @@ func GetPodName(info *cadvisorapiv2.ContainerInfo) string {
 	return info.Spec.Labels[kubernetesPodNameLabel]
 }
 
+func GetPodUID(info *cadvisorapiv2.ContainerInfo) string {
```
Comment (vishh) [inline pkg/kubelet/dockertools/labels.go]: We should not inject cadvisor API into docker package :(. Not for this PR. but we should go back and clean this up.

File: pkg/kubelet/server/stats/summary_test.go [hunk 2]
```diff
@@ -81,9 +81,9 @@ func TestBuildSummary(t *testing.T) {
 		cName20 = "c1" // ensure cName20 conflicts with cName01, but is in a different pod + namespace
 	)
 
-	prf0 := NonLocalObjectReference{Name: pName0, Namespace: namespace0}
-	prf1 := NonLocalObjectReference{Name: pName1, Namespace: namespace0}
-	prf2 := NonLocalObjectReference{Name: pName2, Namespace: namespace2}
+	prf0 := PodReference{Name: pName0, Namespace: namespace0}
```
Comment (vishh) [inline pkg/kubelet/server/stats/summary_test.go]: nit: Include UID in at-least one of the tests.

**Comment by vishh** [comment]
Just one nit. Otherwise LGTM

File: pkg/kubelet/dockertools/labels.go (hunk 1 above)
Comment (pwittrock) [inline pkg/kubelet/dockertools/labels.go]: Would injecting the labels []string be acceptable?

Comment (timstclair) [inline pkg/kubelet/dockertools/labels.go]: Yeah. I think that would be better. I'm happy to make the change in this PR if you want.

Comment (vishh) [inline pkg/kubelet/dockertools/labels.go]: Yeah map[string]string is fine by me.

Comment (vishh) [inline pkg/kubelet/dockertools/labels.go]: A cleanup will be much appreciated.

Comment (timstclair) [inline pkg/kubelet/dockertools/labels.go]: Done.

File: pkg/kubelet/server/stats/summary_test.go (hunk 2 above)
Comment (timstclair) [inline pkg/kubelet/server/stats/summary_test.go]: Done.

**Pull request merged**